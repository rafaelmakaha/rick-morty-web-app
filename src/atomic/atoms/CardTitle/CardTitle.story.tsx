import React from 'react'
import { Meta, Story } from '@storybook/react'
import CardTitle, { ICardTitle } from '.'

export default {
  title: 'Atoms/CardTitle'
} as Meta

const Template: Story<ICardTitle> = (args) => <CardTitle {...args} />

export const Normal = Template.bind({})

const argTypes = {}

const defaultArgs: Omit<ICardTitle> = {}

Normal.args = { ...defaultArgs }
Normal.argTypes = argTypes

