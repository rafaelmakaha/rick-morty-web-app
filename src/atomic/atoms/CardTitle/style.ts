import styled from 'styled-components'

export const Title = styled.h2`
  color: ${({ theme }) => theme.colors.black};
  align-self: center;
  font-size: 16px;
  margin: 0;
  text-align: center;
  padding: 6px 0px;
`
