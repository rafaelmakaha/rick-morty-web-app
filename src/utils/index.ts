export * from './assignTestId'
export * from './withProviders'
