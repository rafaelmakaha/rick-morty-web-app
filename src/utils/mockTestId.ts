type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
) =>
  process.env.NODE_ENV !== 'production'
    ? {
        'data-testid': `${componentType}_${componentId}`
      }
    : {}

export const mockTestID = (
  componentType: componentType,
  componentId: string
): string => assignTestId(componentType, componentId)['data-testid'] ?? ''
