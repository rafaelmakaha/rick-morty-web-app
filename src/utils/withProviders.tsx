import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../tokens'

export const StorybookProviders: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const providerDecorator = (Story: any) => (
  <StorybookProviders>
    <Story />
  </StorybookProviders>
)
