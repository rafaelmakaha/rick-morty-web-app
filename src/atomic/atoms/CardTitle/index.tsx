import React from 'react'

import { Title } from './style'

export interface ICardTitle {
  testID?: string
  children?: React.ReactNode
}

const CardTitle: React.FC<ICardTitle> = ({
  testID = 'CardTitle',
  children
}) => <Title>{children}</Title>

export default CardTitle
