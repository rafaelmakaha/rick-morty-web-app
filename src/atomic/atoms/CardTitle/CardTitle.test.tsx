import React from 'react'
import { render, cleanup } from '@testing-library/react'
import CardTitle, { ICardTitle } from './index'
import { mockTestID } from '../../../utils'

describe('Atoms/CardTitle', () => {
    const props: ICardTitle = {
        testID: 'CardItem',
        children: 'Some Children'
    }

    const renderComponent = () => render(<CardTitle {...props} />)

    let component = renderComponent()

    beforeEach(() => {
        jest.clearAllMocks()
        cleanup()
        component = renderComponent()
    })

    const el_container = mockTestID('div', props.testID!)


    it(`should render component #${el_container}`, () => {
        const sut = component.findByTestId(el_container)
        expect(sut).toBeTruthy()
    })

    it('should render snapshot', () => {
        expect(component.container.firstChild).toMatchSnapshot()
    })
})
