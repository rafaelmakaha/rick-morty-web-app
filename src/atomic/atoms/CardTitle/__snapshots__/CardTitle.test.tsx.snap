// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/CardTitle should render snapshot 1`] = `
<h2
  class="style__Title-sc-1wrd5mx-0 fqesxN"
>
  Some Children
</h2>
`;
